src/lib/parser.rs

```diff
@@ -108,8 +108,51 @@ impl Parser {
     Ok(parser)
   }
   pub fn next_char(&mut self) -> Option<char> {
+    // get next
     let letter = self.contents.get(self.index)?;
+
+    // define forward slash, newline & astrix
+    let fs: u8 = '/' as u8;
+    let nl: u8 = '\n' as u8;
+    let ast: u8 = '*' as u8;
+
+    // increase index
     self.index += 1;
+
+    // check for forward slash
+    if letter == &fs {
+      // check for next forward slash
+      let next = self.contents.get(self.index)?;
+      if next == &fs {
+        // detected single line comment
+        // loop until newline (comments are not parsed)
+        loop {
+          let next = self.contents.get(self.index)?;
+          self.index += 1;
+          // check for end line
+          if next == &nl {
+            // new line detected (end of comment)
+            break
```

Comment:
There is an issue here i think if `next_char` is called and there is a comment it will always return `Some('/')` here because at the bottom you return the letter found at the top that is in the case of a comment always a `/`  
I think this will solve the issue:
```suggestion
            return self.next_char();
```